core/src/main/scala/org/apache/spark/deploy/ApplicationDescription.scala

```diff
@@ -23,7 +23,9 @@ private[spark] class ApplicationDescription(
     val memoryPerSlave: Int,
     val command: Command,
     var appUiUrl: String,
-    val eventLogDir: Option[String] = None)
+    val sparkVersion: String,
```

Comment:
it's because at this point the `SparkContext` doesn't have an App ID yet. If we want to pass name of the file we'll need to have some place holder (e.g. `{{APP_ID}}_lzf`)